### PR TITLE
Hermes backbone

### PR DESCRIPTION
**Pull request by menivaitsi**
DO NOT MERGE.

### CODE REVIEW

File: AppController/lib/hermes_service.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+# Starts and stops the Hermes (messenger) service.
+module HermesService
+
+  # Starts the Hermes service on this machine. We don't want to monitor
+  # it ourselves, so just tell monit to start it and watch it.
+  def self.start()
+    start_cmd = "/usr/bin/python /root/appscale/Hermes/hermes.py"
+    # The stop command doesn't work and relies on terminate.rb.
+    stop_cmd = "/usr/bin/pkill -9 hermes_service"
+    MonitInterface.start(:hermes, start_cmd, stop_cmd, "9999", {})
```
Comment (nlake44) [inline AppController/lib/hermes_service.rb]: Change the 9999 to the port that hermes operates on.

Comment (nlake44) [inline AppController/lib/hermes_service.rb]: And make that a constant.

**Comment by nlake44** [comment]
You write beautiful code Meni! :+1:

File: AppController/lib/hermes_service.rb (hunk 1 above)
Comment (menivaitsi) [inline AppController/lib/hermes_service.rb]: Yeah I've been wondering about the rest of them that have 9999 appended. I think we should drop it altogether.

File: Hermes/hermes.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+import hermes_constants
+from handlers import MainHandler
+from handlers import PollHandler
+from handlers import TaskHandler
+import helper
+
+# Tornado web server options.
+define("port", default=hermes_constants.HERMES_PORT, type=int)
+define("debug", default=True, type=bool)
```
Comment (cdonati) [inline Hermes/hermes.py]: Is this option desirable for production?

File: Hermes/hermes.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  IOLoop.instance().add_callback(shutdown)
+
+def shutdown():
+  """ Shuts down the server. """
+  logging.info("Hermes shutting down.")
+  IOLoop.instance().stop()
+
+def main():
+  """ Main. """
+  logging.getLogger().setLevel(logging.DEBUG)
```
Comment (cdonati) [inline Hermes/hermes.py]: Is this the standard level we use?

File: Hermes/handlers.py [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    logging.info("Sending request to AppScale Portal.")
+    url = "{0}{1}".format(hermes_constants.PORTAL_URL,
+        hermes_constants.PORTAL_POLL_PATH)
+    data = json.dumps({
+      JSONTags.DEPLOYMENT_ID: helper.get_deployment_id()
+    })
+    request = helper.create_request(url=url, method='POST', body=data)
+    response = helper.urlfetch(request)
+    if not response[JSONTags.DEPLOYMENT_ID]:
+      self.set_status(hermes_constants.HTTP_Codes.HTTP_OK)
```
Comment (cdonati) [inline Hermes/handlers.py]: I haven't seen what the portal is supposed to return, but should this be logged?

Comment (menivaitsi) [inline Hermes/handlers.py]: I removed that check. The next block checks for necessary params. Thanks for pointing out.

**Pull request merged**